### PR TITLE
Proposals branch

### PR DESCRIPTION
**Pull request by robstryker**
## What it does

Splits the proposal logic into core and UI. It moves the vast majority of the logic into the core classes, while maintaining the UI as existing wrappers for stability. 

This patch also creates an IInvocationContextCore class and makes it the default interface for use in the core objects, in the same way as #813 made the core classes use the relevant Core interface. Just as in the problem location situation, the UI's API interface extends the hidden non-API core one. 

This patch depends on #813  and #814 

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by robstryker** [comment]
This patch is very difficult to proof-read, due to the fact that the bulk (but not the entirety) for about 37 classes have been moved into their own new unique internal core classes. This means the vast majority of the patch is removals of 90% of one file, and creation of that same code in a new file. 

I'm not sure how to improve this patch to make it more easily followable and to verify that no logic changes have occurred hidden in the patch. =/

**Comment by jjohnstn** [comment]
Hi @robstryker There are new conflicts to fix for this patch.

**Comment by robstryker** [comment]
Squashed and rebased. Looking for a final green build.

File: org.eclipse.jdt.core.manipulation/proposals/org/eclipse/jdt/internal/ui/text/correction/AddAllMissingJavadocTagsProposalCore.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+package org.eclipse.jdt.internal.ui.text.correction;
```
Comment (jjohnstn) [inline org.eclipse.jdt.core.manipulation/proposals/org/eclipse/jdt/internal/ui/text/correction/AddAllMissingJavadocTagsProposalCore.java:1]: Missing license.  A few others are missing licenses as well.  Other than that, looks good.

**Review by jjohnstn — CHANGES_REQUESTED** [review]

**Comment by jjohnstn** [comment]
@robstryker Please also rebase.  It would be good if you can get this in before Monday where large patches are frowned upon.

**Review by jjohnstn — APPROVED** [review]

**Comment by jjohnstn** [comment]
Jenkins appears to be off so new build is not occurring.  Last change built successfully and the only change since then was the addition of copyright headers done my myself.  This refactoring change should go into M1 as it is too big for M2 or M3 so I am merging this before the Monday deadline which requires no large changes.

**Comment by jjohnstn** [comment]
@iloveeclipse please note above.

**Pull request merged**